src/js/chapter-factory.js

```diff
@@ -0,0 +1,87 @@
+import { makeDraggable, makeDroppable } from "./drag-drop";
+
+//Chapter factory
+const Chapter = (story, images, question, chapterNumber) => {
+  let completed = false;
+
+  const changeStatus = () => (completed = !completed);
+  const getStatus = () => completed;
+
+  return { story, images, question, chapterNumber, changeStatus, getStatus };
+};
+
+//Chapter DOM structure
+const createChapterStructure = (chapterObj) => {
+  const createButtons = () => {
+    const homeButton = document.createElement("button");
+    const nextChapterButton = document.createElement("button");
+
+    homeButton.textContent = "home";
+    homeButton.classList.add("home");
+    nextChapterButton.textContent = "-->";
+    nextChapterButton.classList.add("next");
+
+    return { homeButton, nextChapterButton };
+  };
+
+  const setupImages = () => {
+    const images = [...chapterObj.images];
+    const gallery = document.createElement("div");
+
+    for (let image of images) {
+      const picture = document.createElement("img");
+
+      picture.src = image.url;
+      picture.classList.add(image.cssClass);
+      picture.style.height = "100px";
+      picture.style.width = "100px";
+
+      if (image.cssClass !== "drop-container") {
+        makeDraggable(picture);
+      }
+
+      gallery.append(picture);
+    }
+
+    makeDroppable(
+      //Selector for correct picture,
```

Comment:
What about using always the first image in the passed in array as the one that should be droppable?

Reply:
For me personally using class naming convention is more flexible. This way, images in array can be in any order and we don't need to rember that we have to pass droppable as first image. We can even go a little further, and add for droppable and the right draggable images objects new property with id. Thanks to that, we will know which two images should be passed to makeDroppable.

Reply:
Ok don't fully understand, but sounds like something good. :thumbsup: